core/src/main/scala/tech/beshu/ror/accesscontrol/logging/AuditingTool.scala

```diff
@@ -109,6 +110,15 @@ class AuditingTool(settings: Settings,
     }
   }
 
+  private def findLastBlockContext[B <: BlockContext](forbidden: ResponseContext.Forbidden[B]) =
+    forbidden.history
+      .flatMap(_.items)
+      .map(_.result)
+      .reverse
```

Comment:
I want to mention one thing. Here, I'm getting last `Fulfilled` rule. I see `blockContext` is updated in each `Fulfilled` result, so last `Fulfilled` should contain last, and I believe, and is, or should be proven in test?